### PR TITLE
[EasyErrorHandler] Add Symfony Profiler DataCollector

### PR DESCRIPTION
**Pull request by ERuban**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #585

Add Symfony's Profiler DataCollector to collect reported exceptions to simplify functional tests.

### CODE REVIEW

File: packages/EasyErrorHandler/src/Bridge/Symfony/Interfaces/TraceableErrorHandlerInterface.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+declare(strict_types=1);
+
+namespace EonX\EasyErrorHandler\Bridge\Symfony\Interfaces;
+
+use EonX\EasyErrorHandler\Interfaces\ErrorHandlerInterface;
+use Throwable;
+
+interface TraceableErrorHandlerInterface extends ErrorHandlerInterface
+{
+    public function getReportedException(): ?Throwable;
```
Comment (natepage) [inline packages/EasyErrorHandler/src/Bridge/Symfony/Interfaces/TraceableErrorHandlerInterface.php:12]: @ERuban applications could report more than one error per request, so this should return an array.

Also the class is called "ErrorHandlerInterface" so this method should say "getReportedErrors()".

And last one, what about the rendered errors? Applications could use the render method to render more than one errors without necessarily returning them to the consumers, and we might want to collect them too.

**Review by natepage — CHANGES_REQUESTED** [review]

**Comment by natepage** [comment]
@ERuban also, if we really go down the path of implementing a data collector, then I think we should do properly (template, etc.) might be useful.

That said, it would be great to collect the list of ResponseDataBuilders and Reporters so we can show them in the profiler with their class and priority :wink:

File: packages/EasyErrorHandler/src/Bridge/Symfony/Interfaces/TraceableErrorHandlerInterface.php (hunk 1 above)
Comment (ERuban) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Interfaces/TraceableErrorHandlerInterface.php]: Done

**Review by roman-eonx — APPROVED** [review]

File: packages/EasyErrorHandler/src/Interfaces/ErrorHandlerInterface.php [hunk 2]
```diff
@@ -10,6 +10,16 @@
 
 interface ErrorHandlerInterface
 {
+    /**
+     * @return \EonX\EasyErrorHandler\Interfaces\ErrorResponseBuilderInterface[]
+     */
+    public function getBuilders(): array;
```
Comment (natepage) [inline packages/EasyErrorHandler/src/Interfaces/ErrorHandlerInterface.php:16]: I don't think we should require those methods to be implemented by all ErrorHanlders when it's only a thing we need for symfony in dev/test environment.

File: packages/EasyErrorHandler/src/ErrorHandler.php [hunk 3]
```diff
@@ -55,6 +55,16 @@ public function __construct(
         $this->isVerbose = $isVerbose ?? false;
     }
 
+    public function getBuilders(): array
```
Comment (natepage) [inline packages/EasyErrorHandler/src/ErrorHandler.php:58]: PHPDoc is missing.

File: packages/EasyErrorHandler/src/ErrorHandler.php [hunk 4]
```diff
@@ -55,6 +55,16 @@ public function __construct(
...
+    {
+        return $this->builders;
+    }
+
+    public function getReporters(): array
```
Comment (natepage) [inline packages/EasyErrorHandler/src/ErrorHandler.php:63]: PHPDoc is missing.

File: packages/EasyErrorHandler/src/Bridge/Symfony/DependencyInjection/Compiler/ApiPlatformCompilerPass.php [hunk 5]
```diff
@@ -5,13 +5,24 @@
...
+use EonX\EasyErrorHandler\Interfaces\ErrorHandlerInterface;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
 
 class ApiPlatformCompilerPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container): void
     {
+        if ($container->hasParameter('kernel.debug') && $container->getParameter('kernel.debug')) {
```
Comment (natepage) [inline packages/EasyErrorHandler/src/Bridge/Symfony/DependencyInjection/Compiler/ApiPlatformCompilerPass.php:19]: This has nothing to do with ApiPlatform, it shouldn't be in this compiler pass.

**Review by natepage — CHANGES_REQUESTED** [review]

File: packages/EasyErrorHandler/src/Bridge/Symfony/DependencyInjection/Compiler/ApiPlatformCompilerPass.php (hunk 5 above)
Comment (ERuban) [reply packages/EasyErrorHandler/src/Bridge/Symfony/DependencyInjection/Compiler/ApiPlatformCompilerPass.php:19]: `EasyErrorHandlerSymfonyCompilerPass` naming will be ok here?

File: packages/EasyErrorHandler/src/Interfaces/ErrorHandlerInterface.php (hunk 2 above)
Comment (ERuban) [reply packages/EasyErrorHandler/src/Interfaces/ErrorHandlerInterface.php]: Yeap, agree with you. Done it with `@method` annotation in the `ErrorHandlerInterface` and moving it to the `TraceableErrorHandlerInterface`. What do you think?

File: packages/EasyErrorHandler/src/ErrorHandler.php (hunk 3 above)
Comment (ERuban) [reply packages/EasyErrorHandler/src/ErrorHandler.php:61]: Done

File: packages/EasyErrorHandler/src/ErrorHandler.php (hunk 4 above)
Comment (ERuban) [reply packages/EasyErrorHandler/src/ErrorHandler.php:69]: Done

File: packages/EasyErrorHandler/src/Bridge/Symfony/DependencyInjection/Compiler/ApiPlatformCompilerPass.php (hunk 5 above)
Comment (ERuban) [reply packages/EasyErrorHandler/src/Bridge/Symfony/DependencyInjection/Compiler/ApiPlatformCompilerPass.php]: Done it with `ErrorHandlerCompilerPass` name

**Pull request merged**